### PR TITLE
Add MongoDB auto-discovery

### PR DESCRIPTION
**Pull request by nlindblad**

### CODE REVIEW

File: identity-admin-api/app/configuration/Config.scala [hunk 1]
```diff
@@ -11,6 +11,7 @@ object Config {
   val applicationName = "identity-admin-api"
 
   val stage = config.getString("stage")
+  val stack = "identity"
```
Comment (danmassie) [inline identity-admin-api/app/configuration/Config.scala]: Should probably be passed in as the stack is configurable in CF

File: identity-admin-api/app/configuration/MongoConfig.scala [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+          username <- Play.configuration.getString("mongodb.username")
+          password <- Play.configuration.getString("mongodb.password")
+          database <- Play.configuration.getString("mongodb.database")
+        } yield mongoAutoDiscovery.mongoUri(username, password, database, Config.stage, Config.stack)
+        uri.getOrElse(throw new RuntimeException("Could not construct Mongo URI. Missing mongodb.username, mongodb.password or mongodb.database.")).toString
+      }
+    }
+  }
+
+}
```
Comment (danmassie) [inline identity-admin-api/app/configuration/MongoConfig.scala]: New line

File: identity-admin-api/app/repositories/ReactiveMongoConnection.scala [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      addStopHook { () => Future(mongoDriver.close()) }
+    mongoDriver
+  }
+
+  def init(): DefaultDB = {
+    val uri = MongoConfig.uri
+    registerDriverShutdownHook(driver)
+    MongoConnection.parseURI(uri).map { parsedUri =>
+      val connection = driver.connection(parsedUri)
+      val databaseName = parsedUri.db.getOrElse(throw new Exception("No database specified in provided URI"))
```
Comment (danmassie) [inline identity-admin-api/app/repositories/ReactiveMongoConnection.scala]: Very generic exception

File: identity-admin-api/app/configuration/MongoConfig.scala (hunk 2 above)
Comment (nlindblad) [inline identity-admin-api/app/configuration/MongoConfig.scala]: Fixed.

**Pull request merged**